zerver/tornado/event_queue.py

```diff
@@ -1222,20 +1224,11 @@ def get_client_payload(
         message_dict = get_client_payload(
             apply_markdown=client.apply_markdown,
             client_gravatar=client.client_gravatar,
+            allow_empty_topic_name=client.empty_topic_name,
             can_access_sender=can_access_sender,
             is_incoming_1_to_1=wide_dict["recipient_id"] == client.user_recipient_id,
         )
 
-        # Compatibility code to change subject="" to subject=Message.EMPTY_TOPIC_FALLBACK_NAME
-        # for older clients with no UI support for empty topic name.
-        if (
-            recipient_type_name == "stream"
-            and not client.empty_topic_name
-            and wide_dict["subject"] == ""
-        ):
-            message_dict = message_dict.copy()
-            message_dict["subject"] = Message.EMPTY_TOPIC_FALLBACK_NAME
```

Comment:
I'm a little confused by why the second commit removes this.

Reply:
`get_client_payload` makes a call to `MessageDict.finalize_payload` which performs this logic. We can stash the two commits -- kept it separate as the changes will be a lot to review in one commit.